src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -0,0 +1,124 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.sps.data.Meal;
+
+@WebServlet("/meal/*")
+public class MealServlet extends HttpServlet {
+
+    private ArrayList<Meal> dishes;
+
+    @Override
+    public void init() {
+        dishes = new ArrayList<>(Arrays.asList());
+        dishes.add(new Meal(0, "fried potato", "fried potato with mushrooms and onion",
+            new ArrayList<>(Arrays.asList("potato", "onion", "mushrooms", "oil")), "Main"));
+        dishes.add(new Meal(1, "Italian pizza", "Pizza with pineaple, sausage and tomato",
+            new ArrayList<>(Arrays.asList("flour", "water", "sausage", "tomato", "pineaple", "cheese")), "pizza"));
+        dishes.add(new Meal(2, "vegetable soup", "vegetable soup with onion",
+            new ArrayList<>(Arrays.asList("potato", "onion", "cabbage", "mushrooms", "water", "carrot", "pumpkin")), "soup"));
+        dishes.add(new Meal(3, "chocolate cake", "chocolate cake with butter cream and strawberry",
+            new ArrayList<>(Arrays.asList("flour", "water", "butter", "cacao", "chocolate", "sugar", "strawberry", "eggs")), "dessert"));
+        dishes.add(new Meal(4, "hot chocolate", "hot chocolate with sugar and caramel",
+            new ArrayList<>(Arrays.asList("cacao", "sugar", "milk", "caramel", "vanil")), "drinks"));
+    }
+
+    /*@Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String gson = new Gson().toJson(result);
+
+        response.setContentType("application/json");
+        response.getWriter().println(gson);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        result = searchDishByKeyWord(getParameter(request, "text", ""));
+        
+        response.setContentType("application/json;");
+        response.getWriter().println(result);
+        response.sendRedirect("/index.html");
+    }
+    
+    private ArrayList<String> searchDishByKeyWord(String requestedDish) {
+        ArrayList<String> result = new ArrayList<String>();
+
+        for (String dish : this.dishes) {
+            if (dish.contains(requestedDish)) {
+                result.add(dish);
+            }
+        }
+        return result;
+    }
+*/
+
+    // Get meal
+	// GET/meal/
+	// GET/meal/Id
+    @Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		String pathInfo = request.getPathInfo();
+
+		if(pathInfo == null || pathInfo.equals("/")){
```

Comment:
Nit:
Probably it will be a good idea to add comments that describe proceeded paths right next to the pieces of code that do this stuff.
For example
```
// GET/meal/ or empty path
if (pathInfo == null || pathInfo.equals("/")) {
...
}
```